src/react/account/AccountUserList.tsx

```diff
@@ -104,49 +118,90 @@ const renderActionButtons = (rowValues) => {
     <div style={{ display: 'flex' }}>
       <CopyARNButton text={arn} />
       <RenderEditButton userName={userName} />
+      <DeleteUserAction userName={userName} />
     </div>
   );
 };
 
-const WithTooltipWhileLoading = ({
-  children,
-  isLoading,
-  tooltipOverlay,
-}: {
-  tooltipOverlay: string;
-  isLoading?: boolean;
-  children: JSX.Element;
-}) => (
-  <>
-    {isLoading ? (
-      <Tooltip overlay={tooltipOverlay}>{children}</Tooltip>
-    ) : (
-      children
-    )}
-  </>
-);
+const DeleteUserAction = (rowValue: { userName : string} , accountName: string) => {
+  const { userName } = rowValue;
+  const IAMClient = useIAMClient();
+  const [showModal, setShowModal] = useState(false);
+  const { data: accessKeysResult, status: accessKeyStatus } =
+    useAwsPaginatedEntities(
+      getUserAccessKeysQuery(userName, notFalsyTypeGuard(IAMClient)),
+      (data) => data.AccessKeyMetadata,
+    );
+  const { data: listGroupsResult, status: listGroupStatus } = useQuery(
+    getUserListGroupsQuery(userName, notFalsyTypeGuard(IAMClient)),
+  );
 
-const SEARCH_QUERY_PARAM = 'search';
+  const deleteUserMutation = useMutation(
+    (userName: string) => {
+      return notFalsyTypeGuard(IAMClient).deleteUser(userName);
+    },
+    {
+      onSuccess: () =>
+        queryClient.invalidateQueries(
+          getUserListUsersQuery(accountName, notFalsyTypeGuard(IAMClient))
+            .queryKey,
+        ),
+    },
+  );
+
+  return (
+    <>
+      <DeleteConfirmation
+        show={showModal}
+        cancel={() => setShowModal(false)}
+        approve={() => {
+          deleteUserMutation.mutate(userName);
+        }}
+        titleText={`Permanently remove the following user ${userName} ?`}
+      />
+
+      <Button
+        id="delete-accessKey-btn"
+        disabled={
+          (accessKeysResult && accessKeysResult?.length >= 1) ||
+          (listGroupsResult && listGroupsResult.Groups?.length >= 1) ||
+          accessKeyStatus === 'loading' ||
+          listGroupStatus === 'loading'
+        }
+        icon={<i className="fas fa-trash" />}
+        style={{ height: spacing.sp24, marginLeft: '0.6rem' }}
+        label="Delete"
+        onClick={() => {
+          setShowModal(true);
+        }}
+        variant="danger"
+        tooltip={{
+          overlay:
+            accessKeyStatus === 'loading' ? 'loading...' : 'Remove accessKey',
+          placement: 'right',
+        }}
+      />
+      {accessKeyStatus === 'error' && (
+        <Banner
+          icon={<i className="fas fa-exclamation-triangle" />}
+          title="Error: Unable to delete user"
+          variant="danger"
+        >
+          Error: Unable to delete user.
+        </Banner>
+      )}
+    </>
+  );
+};
```

Comment:
I believe the feature of Delete User is backportted from development/1.5. 
In term of good practice, I would put them in a dedicated commit.